src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts

```diff
@@ -81,19 +82,106 @@ export class InstructorCourseDetailsPageComponent implements OnInit {
   }
 
   /**
-   * Loads the course's details based on the given course ID and email.
+   * Loads the course's details based on the given course ID.
    */
   loadCourseDetails(courseid: string): void {
-    const paramMap: { [key: string]: string } = { courseid };
-    this.httpRequestService.get('/courses/details', paramMap).subscribe((resp: CourseInfo) => {
-      this.courseDetails = resp.courseDetails;
-      this.currentInstructor = resp.currentInstructor;
-      this.instructors = resp.instructors;
-      this.sections = resp.sections;
-
-      if (!this.courseDetails) {
-        this.statusMessageService.showErrorMessage('Error retrieving course details');
-      }
+    this.loadCourseName(courseid);
+    this.loadInstructors(courseid);
+    this.loadStudents(courseid);
+  }
+
+  /**
+   * Loads the name of the course
+   */
+  private loadCourseName(courseid: string): void {
+    this.courseService.getCourseAsInstructor(courseid).subscribe((course: Course) => {
+      this.courseDetails.course = {
+        id: courseid,
+        name: course.courseName,
+      };
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorMessage(resp.error.message);
+    });
+  }
+
+  /**
+   * Loads the instructors in the course
+   */
+  private loadInstructors(courseid: string): void {
+    const paramMap: any = { courseid, intent: Intent.FULL_DETAIL };
+    this.httpRequestService.get('/instructors', paramMap).subscribe((instructors: Instructors) => {
+      this.instructors = instructors.instructors.map((instructor: Instructor) => {
+        return {
+          email: instructor.email,
+          name: instructor.name,
+          role: instructor.role,
+        };
+      });
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorMessage(resp.error.message);
+    });
+  }
+
+  /**
+   * Loads the students in the course
+   */
+  private loadStudents(courseid: string): void {
+    this.studentService.getStudentsFromCourse(courseid).subscribe((students: Students) => {
+      const sections: StudentIndexedData = students.students.reduce((acc: StudentIndexedData, x: Student) => {
```

Comment:
There are better way to count the number of sections and teams? Use `Set()`?

Reply:
For sections, it's not only used to count the number of sections but to also group the students into sections. But for teams then yes, I'll update it to use `Set`.

By the way, `loadStudents` and `loadPrivilege` were borrowed from `instructor-student-list-page`

Reply:
> For sections, it's not only used to count the number of sections but to also group the students into sections. 

You should remember long time ago, you are trying to flat the student list but fail :P So now you see how inconvenient it is.